"No new tests" text has no newline after it when created via commit message hook

hortont424 opened the pull request:
#### eb70a062e9c700eecec9c3fdcea6f85e6cfd8f41
<pre>
&quot;No new tests&quot; text has no newline after it when created via commit message hook
<a href="https://bugs.webkit.org/show_bug.cgi?id=299534">https://bugs.webkit.org/show_bug.cgi?id=299534</a>
<a href="https://rdar.apple.com/161340860">rdar://161340860</a>

Reviewed by Jonathan Bedard and Sam Sneddon.

* Tools/Scripts/hooks/prepare-commit-msg:
After 299869@main, the output of prepare-ChangeLogs now includes important newlines
(specifically, between &quot;no new tests&quot; and the list of changed files).

The existing code in prepare-commit-msg strips all empty lines.

I initially started by trying to teach it to preserve this line, but then realized
that all of this complexity is purely here to serve merging multiple directories&apos;
ChangeLog&apos;s file lists together, which is no longer a concern now that we have
no physical ChangeLogs. (There is yet more complexity in both this script
and prepare-ChangeLogs itself that can be cleaned up now that this is true.) So,
instead, just delete this code, and don&apos;t have prepare-ChangeLogs add delimiters,
and maintain the formatting that it outputs.

Canonical link: <a href="https://commits.webkit.org/300525@main">https://commits.webkit.org/300525@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a1a7cd53a4a54a29b551c4bb50ffe191f0381e62

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122932 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/42646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33343 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129579 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/75040 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/089e12a5-9d7a-4efc-a807-d10fae5c9a5b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43369 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51240 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/93455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/75040 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/93a921ca-137c-48b5-bf68-52f1342c0639) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125883 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/34577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110035 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/74072 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8e9a5443-67aa-4a47-b6f1-892af007f9cd) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/33557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28189 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73076 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/104283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28400 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132306 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/49881 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/37975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/101953 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/50258 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106241 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/101819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25859 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25371 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/49737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/49204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/52556 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/50886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (approved):
I think this is right.

We really should go back to trying to make this an actual valid Python script syntactically and then actually write tests for this. There's just entirely too much logic in here for my liking. And then I can judge people for "no new tests" when they fix bugs like this. 🙂

hortont424 committed:
"No new tests" text has no newline after it when created via commit message hook
https://bugs.webkit.org/show_bug.cgi?id=299534
rdar://161340860

Reviewed by Jonathan Bedard and Sam Sneddon.

* Tools/Scripts/hooks/prepare-commit-msg:
After 299869@main, the output of prepare-ChangeLogs now includes important newlines
(specifically, between "no new tests" and the list of changed files).

The existing code in prepare-commit-msg strips all empty lines.

I initially started by trying to teach it to preserve this line, but then realized
that all of this complexity is purely here to serve merging multiple directories'
ChangeLog's file lists together, which is no longer a concern now that we have
no physical ChangeLogs. (There is yet more complexity in both this script
and prepare-ChangeLogs itself that can be cleaned up now that this is true.) So,
instead, just delete this code, and don't have prepare-ChangeLogs add delimiters,
and maintain the formatting that it outputs.

Canonical link: https://commits.webkit.org/300525@main
